qutebrowser/mainwindow/mainwindow.py

```diff
@@ -123,6 +123,8 @@ class MainWindow(QWidget):
     Attributes:
         status: The StatusBar widget.
         tabbed_browser: The TabbedBrowser widget.
+        last_non_fullscreen_window_state: window state to restore when going out of fullscreen by :fullscreen
+        last_window_state: window state to restore when _on_fullscreen_requested(false) is called
```

Comment:
What about `window_state_before_fullscreen` and `window_state_before_page_fullscreen`?